honeybadger-io/honeybadger-js
# #654 Always call afterNotify

## Status
**READY**

## Description
Issue #654.
Make sure to always call `beforeNotify` and `afterNotify` handlers, even if report is not sent to Honeybadger.

## Todos
- [x] Always call `afterNotify`
- [x] Always call `beforeNotify`
- [x] Lambda functions: should the callback always be called in `afterNotify`? (as it was before #650)
- [x] Tests (`afterNotify` in notice, multiple `afterNotify`, `afterNotify` in notice and in client instance, `afterNotify` in lambda functions)

## Steps to Test or Reproduce
1. Create an HB instance without setting `apiKey`.
2. Set an `afterNotify` handler: client.afterNotify((err) => console.log(err.message))
3. Call client.notify('should not send report')
4. The `afterNotify` handler should be called with an error saying that `apiKey` was not set.


<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_3u7vl)
